app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java

```diff
@@ -1,55 +1,227 @@
+     */
     public static void goToUserProfileActivity(View view) {
         Context context = view.getContext();
         Intent intent = new Intent(context, ProfileActivity.class);
         context.startActivity(intent);
     }
 
+    /**
+     * Starts the MainActivity activity
+     * @param view the View in which to start the activity
+     */
     public static void goToHome(View view) {
         Context context = view.getContext();
         Intent intent = new Intent(context, MainActivity.class);
         context.startActivity(intent);
     }
 
+    /**
+     * Starts the chat activity
+     * @param view the View in which to start the activity
+     */
     public static void goToChat(View view) {
         Context context = view.getContext();
         Intent intent = new Intent(context, ChatActivity.class);
         intent.putExtra("ACTIVITY_CHAT_ID", "general_chat");
         context.startActivity(intent);
     }
 
+    /**
+     * Starts ActivityList activity
+     * @param view the View in which to start the activity
+     */
     public static void goToListOfActivities(View view) {
         Context context = view.getContext();
         Intent intent = new Intent(context, ActivityListActivity.class);
         context.startActivity(intent);
     }
 
+
+    /**
+     * Fills in the navigation bar's text fields with the user's information
+     * 0 = fullName
+     * 1 = email
+     * 2 = phone number
+     * 3 = description (optional)
+     * @param fields the fields to fill in, following the above order
+     * @return the array of updated TextViews
+     */
+    public static TextView[] fillNavigationProfileFields(TextView[] fields) {
+        FirebaseUser firebaseUser = AuthenticationInstanceProvider.getAuthenticationInstance().getCurrentUser();
+        if (firebaseUser != null) {
+            String userId = firebaseUser.getUid();
+            UserSerializer serializer = new UserSerializer();
+            FirestoreUserManager backendUserManager =
+                    new FirestoreUserManager(BackendInstanceProvider.getFirestoreInstance(),
+                            FirestoreUserManager.USERS_COLLECTION,
+                            serializer);
+            backendUserManager.getUserFromUid(userId).addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
+                @Override
+                public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                    if (task.isSuccessful()) {
+                        User user = serializer.deserialize(task.getResult().getData());
+
+                        fields[0].setText(user.getFullName());
+                        fields[1].setText(user.getEmail());
+                        fields[2].setText(user.getPhoneNumber());
+                        if (fields.length > 3 && user.getDescription() != null && !user.getDescription().isEmpty())
+                            fields[3].setText(user.getDescription());
```

Comment:
Filling in the "user description" in the navigation bar wouldn't fill it too much?

Reply:
I wasn't sure where else this function was meant to be used, so I copied the description part as well so that it may be used elsewhere in the future.

Reply:
Okay! This will be something new then (but why not)! Because for the moment I guess that only the name, the phone number and the email address of the user (but not his description) are displayed in the navigation bar.